auem/data/datasets/audioset.py

```diff
@@ -111,33 +156,41 @@ def __init__(
         }
         self.l2c = {v: k for k, v in self.c2l.items()}
 
-        datapaths = glob(audioset_path + "*[m4a|webm]")
-        datapaths = {os.path.splitext(os.path.basename(x))[0]: x for x in datapaths}
-
-        self.annotations = []
-        with open(audioset_annotations, "r") as f:
-            for row in AudiosetAnnotationReaderV1(f, ontology=self.ontology):
-                if row["YTID"] in datapaths.keys():
-                    self.annotations.append(
-                        {
-                            "ytid": row["YTID"],
-                            "start_seconds": row["start_seconds"],
-                            "end_seconds": row["end_seconds"],
-                            "classes": [
-                                self.classes.index(k)
-                                for k in row["positive_labels"].keys()
-                            ],
-                            "filepath": datapaths[row["YTID"]],
-                        }
-                    )
+        audioset_path = Path(audioset_path)
+        self.datapaths = {x.stem: x for x in audioset_path.glob("*[m4a|webm]")}
+
+        self.annotations = AudiosetAnnotationReaderV2(
+            audioset_annotations, classes=self.classes
+        )
+        # for row in AudiosetAnnotationReaderV1(f, ontology=self.ontology):
```

Comment:
See my comment above, removing this also breaks compatability with V1 :(

Reply:
also ditto what i said above